src/nat/builder/function.py

```
@@ -342,3 +344,178 @@ class FunctionImpl(LambdaFunction[input_type, streaming_output_type, single_outp
+        FunctionGroupBaseConfig
+            The configuration for the function group.
+        """
+        return self._config
+
+    def _get_all_but_excluded_functions(self) -> dict[str, Function]:
+        """
+        Returns a dictionary of all functions in the function group except the excluded functions.
+        """
+        if set(self._config.exclude) - set(self._functions.keys()):
+            raise ValueError(f"Function group {self._instance_name} excludes functions that are not found in the group."
+                             f" Available functions: {list(self._functions.keys())}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)